spec/features/work_packages/table/project_phase_field_spec.rb

```diff
@@ -0,0 +1,172 @@
+           })
+  end
+  let!(:query) do
+    build(:public_query, user: current_user, project: work_package.project)
+  end
+  let(:query_columns) { %w(subject project_phase) }
+
+  current_user { user }
+
+  before do
+    query.column_names  = query_columns
+    query.sort_criteria = sort_criteria if sort_criteria
+    query.group_by      = group_by if group_by
+    query.filters.clear
+    query.show_hierarchies = false
+    query.save!
+
+    wp_table.visit_query query
+    wp_table.expect_work_package_listed work_package
+
+    wait_for_network_idle
+  end
+
+  context "with the feature flag being active", with_flag: { stages_and_gates: true } do
+    context "with the phase being active" do
+      it "shows the project phase column with the correct phase for the work package" do
+        wp_table.expect_work_package_with_attributes(work_package, { projectPhase: project_phase.name })
+        wp_table.expect_work_package_with_attributes(other_wp, { projectPhase: other_project_phase.name })
+        wp_table.expect_work_package_with_attributes(wp_without_phase, { projectPhase: "-" })
+      end
+
+      context "when sorting by project phase ASC" do
+        let(:sort_criteria) { [%w[project_phase asc]] }
+
+        it "sorts ASC by phase position" do
+          wp_table.expect_work_package_order(other_wp, work_package, wp_without_phase)
+        end
+      end
+
+      context "when sorting by project phase DESC" do
+        let(:sort_criteria) { [%w[project_phase desc]] }
+
+        it "sorts DESC by phase position" do
+          wp_table.expect_work_package_order(wp_without_phase, work_package, other_wp)
+        end
+      end
+
+      context "when editing the value of a project phase cell" do
+        it "changes the value" do
+          wp_table.update_work_package_attributes(wp_without_phase, projectPhase: project_phase_definition)
+          wp_table.expect_work_package_with_attributes(wp_without_phase, { projectPhase: project_phase.name })
+        end
+      end
+
+      context "when grouping by project phase" do
+        let(:group_by) { :project_phase }
+
+        it "groups by project phase" do
+          wp_table.expect_groups({
+                                   project_phase.name => 1,
+                                   other_project_phase.name => 1,
+                                   "-" => 1
+                                 })
+        end
+
+        it "includes the group icon in the group row header" do
+          within("#wp-table-rowgroup-0") do
+            expect(page).to have_test_selector("project-phase-icon #{other_project_phase.name}")
+          end
+
+          within("#wp-table-rowgroup-1") do
+            expect(page).to have_test_selector("project-phase-icon #{project_phase.name}")
+          end
+        end
+      end
+    end
+
+    context "with one phase being inactive" do
+      let(:other_project_phase) { create(:project_phase, active: false) }
+
+      it "does not show the inactive phase" do
```

Comment:
The grouping of an inactive phase still shows up, I'm not sure what would be the right behaviour.
In the following picture, the "Planning" phase is made inactive:
<img width="1003" alt="image" src="https://github.com/user-attachments/assets/3e1f2561-2cdc-42ed-bba2-e354c9e135f0" />

Reply:
Thanks for bringing this up, very good point! We discussed this in the weekly and we decided that a group with no active phases for the listed work packages should be hidden/shown as "-" in the group header.

I will adjust this PR accordingly :)

Reply:
It was a bit more complicated than I anticipated, but this feedback has been addressed now. I also added a test case for it.

Reply:
Thanks Tobi, I checked now and it displays correctly. However the sorting still takes the omitted definition's position into account. In the screenshot below, "Fifth dev" should be at the bottom of the page, as the second page continues with non phase associated work packages.
<img width="1006" alt="image" src="https://github.com/user-attachments/assets/e8e63965-b315-458b-bb72-1e055c780db5" />
<img width="997" alt="image" src="https://github.com/user-attachments/assets/f9888d12-22d5-4141-9391-bfb3c3f655d3" />

Reply:
I could not reproduce this exact issue 🤔 In your example, is "fifth dev" associated to an deactivated phase?

With a similar setup, I get correctly sorted groups.
![image](https://github.com/user-attachments/assets/6650b1b6-0d3e-4d62-9ceb-dd97238274e6)

While investigating, I found another issue with the sorting that is fixed as of the latest commit. I doubt that it has any effect on the erroneous behavior from your screenshot, but if you have time I'd appreciate it if you took a look. 🙇🏻‍♂️

Reply:
As I re-seeded my data, and I cannot reproduce the issue anymore, it is possible that I had some inconsistent data.